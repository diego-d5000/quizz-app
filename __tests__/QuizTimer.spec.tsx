import React from 'react';
import { render, act, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import QuizTimer from '../components/QuizTimer';

describe('QuizTimer Component', () => {
  const mockDefaultTimer = 6;
  const mockOnFinishCb = jest.fn();

  it('snapshot', () => {
    const tree = renderer.create(<QuizTimer onFinish={mockOnFinishCb} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('starts with default timer', () => {
    const TestingComponent = render(
      <QuizTimer onFinish={mockOnFinishCb} defaultTimer={mockDefaultTimer} />
    );
    const { getByText } = TestingComponent;
    const timeIndicatorElement = getByText(mockDefaultTimer.toString());
    expect(timeIndicatorElement).toBeInTheDocument();
  });
});
