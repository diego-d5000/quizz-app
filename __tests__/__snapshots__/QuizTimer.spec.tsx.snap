// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizTimer Component snapshot 1`] = `
<div
  className="relative h-10 w-full rounded-full overflow-hidden"
>
  <div
    className="text-white font-black left-0 w-full top-0 h-full text-center py-2 absolute z-10"
    style={Object {}}
  >
    <span>
      10
    </span>
  </div>
  <div
    className="absolute left-0 w-full top-0 h-full bg-gray-500 opacity-25"
  />
  <div
    className="absolute left-0 top-0 h-full transition-all ease-out duration-1000 bg-blue-400"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;
